File: src/main/cs/Problem0020.cs
using System;
using System.Diagnostics;
using System.Numerics;
using Euler.test.cs;

namespace Euler.main.cs
{
    public class Problem0020
    {
        public Problem0020()
        {
            Stopwatch stopWatch = new Stopwatch();
            stopWatch.Start();
            BigInteger result = 1;

            // Solution:
            for (int i = 2; i <= 100; i++)
            {
                result *= i;
            }

            String s = result.ToString();
            result = 0;
            foreach (char c in s)
            {
                result += c - 48;
            }


            stopWatch.Stop();
            string elapsedTime = stopWatch.Elapsed.ToString();
            Console.WriteLine("Result:\t" + result + "\tTime:\t" +
                              (double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1, 2)) >= 1
                                  ? double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1)) + " s"
                                  : double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000 +
                                    " ms"));
            if (Test.DoBenchmark)
            {
                Benchmark.AddTime(20, double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000);
            }
        }
    }
}

File: src/main/java/_00/Problem0020.java
package _00;

import java.math.BigInteger;

public class Problem0020 {
    public Problem0020() {
        long startTime = System.nanoTime();
        BigInteger result = BigInteger.ONE;

        // Solution:
        for (int i = 2; i <= 100; i++) {
            result = result.multiply(new BigInteger(i + ""));
        }
        String s = result.toString();
        result = BigInteger.ZERO;
        for (char c : s.toCharArray()) {
            result = result.add(new BigInteger(c - 48 + ""));
        }

        long timeToResolve = System.nanoTime() - startTime;
        System.out.println("Result:\t" + result + "\tTime:\t" + (((double) timeToResolve / 1_000_000) > 1000 ?
                (((double) timeToResolve / 1_000_000_000) + "s") :
                (((double) timeToResolve / 1_000_000) + "ms")));
    }
}
